community/kernel/src/test/java/org/neo4j/unsafe/impl/batchimport/input/GroupsTest.java

```diff
@@ -50,4 +52,109 @@ public void shouldHandleConcurrentGetOrCreate() throws Throwable
         Group otherGroup = groups.getOrCreate( "MyOtherGroup" );
         assertEquals( 1, otherGroup.id() );
     }
+
+    @Test
+    public void shouldFailOnMixedGroupModeInGetOrCreate() throws Exception
+    {
+        // given
+        Groups groups = new Groups();
+        groups.getOrCreate( null );
+
+        try
```

Comment:
Would prefer to have expected exceptions in the added tests since nothing is invoked after the throw.